docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
+>
+> While this model is arguably closer to real life, the program logic would have to operate on different object types depending on whether a given `Task` belongs to a user-created Module or the default General Tasks list. This was deemed to increase coupling and introduce too much unnecessary clutter to the code, hence it was not used.
+
+<br>
+
+### 5.4. Command Component
+
+The `Command` component is in charge of actually executing the operations requested by the user.
+
+The following partial class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/CommandClassDiagram.puml)
+
+All commands inherit the abstract `Command` class and must contain an `execute()` method. The program logic that must be executed to fulfil the requested command is implemented in this method. Additionally, `execute()` returns any command output to be displayed to the user as feedback.
+
+`Command` instances are generated by their corresponding `Parser` classes (e.g. `AddCommand` is constructed by `AddParser`) and executed by `Main`.
+
+<br>
+
+### 5.5. Storage Component
+
+The `Storage` component is responsible for the saving and loading of program data from and to its data files. The following class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Storage.puml)
+
+Storage is an interface supporting write/read data to/from computer storage:
+* Storage interface is implemented by JsonStorage in Mod Happy, which will read and load data to and from json format.
+* ListStorage can save a ArrayList of any class that extends Object in json format, and read them back into corresponding objects. (E.g. ModuleListStorage, TaskListStorage inherit from ListStorage)
+* There are navigability to Storage from Main and SaveCommand, which handles the load and write data to/from disk respectively.
+  <br><br><br>
+
+## 6. Implementation
+
+This section describes some details on how some features are implemented.
+<br>
+
+### 6.1. Edit Feature
+
+The edit feature allows the user to change a parameter of a task/module. The parameters of a module is its module description while the parameters of a task are its task name, task description and estimated working time. 
+
+The following sequence diagram illustrates the process:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/Edit.puml)
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/GetModule.puml)
```

Comment:
![image](https://user-images.githubusercontent.com/65173208/161184194-d92ff3bb-42b3-457f-ad57-ab1f708da324.png)

Really like that you put some methods into a separate ref frame! It makes the diagram a lot clearer. For the condition in the alt frames, I wonder if it would be better to say something like [taskmodule == null] instead? It might be shorter and more readable that way.